tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -21,16 +21,73 @@
 #include "llvm/Support/Path.h"
 #include "llvm/Support/raw_ostream.h"
 
+namespace {
+spv::ExecutionModel getSpirvShaderStageFromHlslProfile(const char *profile) {
+  // DXIL Models are:
+  // Profile (DXIL Model) : HLSL Shader Kind : SPIR-V Shader Kind
+  // vs_<version>         : Vertex Shader    : Vertex Shader
+  // hs_<version>         : Hull Shader      : Tassellation Control Shader
+  // ds_<version>         : Domain Shader    : Tessellation Evaluation Shader
+  // gs_<version>         : Geometry Shader  : Geometry Shader
+  // ps_<version>         : Pixel Shader     : Fragment Shader
+  // cs_<version>         : Compute Shader   : Compute Shader
+  switch (profile[0]) {
```

Comment:
This string check will later be replaced by using `hlsl::ShaderModel`: https://github.com/google/DirectXShaderCompiler/commit/a64deb5616d69a57890379c06ed3b82daf147433